Consolidate Docs guide with original home page (#2453)
ros2/ros2_documentation: source/index.rst

   Related-Projects
   Glossary

**The Robot Operating System (ROS) is a set of software libraries and tools for building robot applications.**
From drivers and state-of-the-art algorithms to powerful developer tools, ROS has the open source tools you need for your next robotics project.

Since ROS was started in 2007, a lot has changed in the robotics and ROS community.
The goal of the ROS 2 project is to adapt to these changes, leveraging what is great about ROS 1 and improving what isn’t.

> For ROS 1 documentation, check out the `ROS wiki <https://wiki.ros.org>`__.

Reviewer: I think we should be a little more explicit that all documentation for ROS 2 is on this site, and that ROS 1 is on the wiki.  Something like:

```suggestion
This site contains the documentation for ROS 2.
If you are looking for ROS 1 documentation, check out the `ROS wiki <https://wiki.ros.org>`__.
```

Author: My thinking was that the title of the page (`ROS 2 Documentation`) would be self-explanatory (as well as the breakdown later of what sites cover which info), but happy to add this edit if you feel strongly about it.

A recurring issue I came across while editing the content for this PR was that the same info was repeated in different ways across different paragraphs, pages, and sites. This made it confusing to determine the source of truth. I didn't know which sources were accurate, up-to-date, or deprecated. I think less repetition, especially in one centralized place like this page, will actually be clearer.

Reviewer: > My thinking was that the title of the page (ROS 2 Documentation) would be self-explanatory (as well as the breakdown later of what sites cover which info), but happy to add this edit if you feel strongly about it.

What I've learned is that everything I think is self-explanatory is confusing to someone :).  Adding in that little bit of documentation will keep us from getting bug reports like "I want to install ROS 1, but there are no instructions here on how to do that".

> A recurring issue I came across while editing the content for this PR was that the same info was repeated in different ways across different paragraphs, pages, and sites. This made it confusing to determine the source of truth. I didn't know which sources were accurate, up-to-date, or deprecated. I think less repetition, especially in one centralized place like this page, will actually be clearer.

Yes, agreed.  That is a problem across the entire docs site.  This documentation site has been written over about 7 years by different people, at different times.  We also consolidated multiple documentation sites together at some point.  So all of it needs a refresh, and a look over, which is why I'm glad you are doing this.

Getting started
---------------

* :doc:`Installation <Installation>`

  - Instructions to set up ROS 2 for the first time

* :doc:`Tutorials <Tutorials>` 

>   - Best place to start for new users!

Reviewer: ```suggestion
  - The best place to start for new users!
```

  - Hands-on sample projects that help you build a progression of necessary skills

* :doc:`How-To-Guides` 

  - Quick answers to your "How do I...?" questions without working through the :doc:`Tutorials <Tutorials>`

* :doc:`Concepts <Concepts>` 

  - High-level explanations of core ROS 2 concepts covered in the :doc:`Tutorials <Tutorials>`

* :doc:`Contributing <Contributing>`

>   - Best practices for contributing to these docs

Reviewer: This isn't actually true; the Contributing page is about Contributing to ROS 2 in general.  So maybe something more like:

```suggestion
  - Best practices and methodology for contributing to ROS 2 as well as migrating existing content from ROS 1 to ROS 2
```

* :doc:`Contact <Contact>` 

  - Answers to your questions or a forum to start a discussion

The ROS 2 project
-----------------

If you're interested in the advancement of the ROS 2 project:

* :doc:`Releases <Releases>` 

  - Past, present and future ROS 2 distributions

* :doc:`Features Status <Features>` 

  - Features in the current release

* :doc:`Feature Ideas <Feature-Ideas>` 

>   - Ideas for features that are coming soon

Reviewer: The "Feature Ideas" are things that would be nice to have, but have no active plans.  So we should say something like:

```suggestion
  - Ideas for features that would be nice to have, but are not currently being worked on
```

* :doc:`Roadmap <Roadmap>` 

  - Planned work for ROS 2 development

* :doc:`ROSCon talks <ROSCon-Content>` 

>   - Presentations by the ROS community on the latest utilizations and updates

Reviewer: I know you reused this wording from above, but I don't know what "latest utilizations" means.  Maybe we should reword this to:

```suggestion
  - Presentations by the ROS community on ROS 2
```

  - Indexed list of all packages (i.e. `Python Package Index (PyPi) <https://pypi.org/>`_ for ROS packages)
  - See which ROS distributions a package supports
  - Link to a package's repository, API documentation, or website
  - Inspect a package's license, build type, maintainers, status, and dependencies
  - Get more info for a package on `ROS Answers <https://answers.ros.org/questions/>`__

* `ROS Prerelease <http://prerelease.ros.org/>`__ (ROS 1)

  - Generates commands to emulate the `ROS Buildfarm <https://build.ros.org/>`_ on your local machine
  - Currently only shows ROS 1 distributions

* `ROS Robots <https://robots.ros.org/>`__ (ROS 1, ROS 2)

  - Showcases robots proects from the community
  - Instructions on how to contribute a robot

* `ROS Wiki <http://wiki.ros.org/>`__ (ROS 1)

  - ROS 1 documentation and user modifiable content
  - Active until at least the last ROS 1 distribution is EOL

* `ROS.org <https://www.ros.org/>`__ (ROS 1, ROS 2)

  - ROS 1 and ROS 2 product landing page, with high-level description of ROS and links to other ROS sites

Deprecated
^^^^^^^^^^

* `ROS Design <http://design.ros2.org/>`__ (ROS 2)

  - Early design decisions

>   - Deprecated in favor of `ROS Enhancement Proposals (REPs) <https://ros.org/reps/rep-0000.html>`__

Reviewer: This is a duplicate with what is up above, so let's just drop this section completely.

Author: Do you mean the entire "Deprecated" section? Or just the "ROS Design" section?

For either case, I'm worried about users stumbling upon those sites while they are still up and active, and being confused as to whether the information there is relevant / accurate / maintained.

Reviewer: > Do you mean the entire "Deprecated" section? Or just the "ROS Design" section?

Just the "ROS Design" section.  We already are linking to it in "The ROS 2 Project" section above.

Getting started
---------------

* :doc:`Installation <Installation>`

  - Instructions to set up ROS 2 for the first time

* :doc:`Tutorials <Tutorials>` 

  - The best place to start for new users!
  - Hands-on sample projects that help you build a progression of necessary skills

* :doc:`How-To-Guides` 

  - Quick answers to your "How do I...?" questions without working through the :doc:`Tutorials <Tutorials>`

* :doc:`Concepts <Concepts>` 

  - High-level explanations of core ROS 2 concepts covered in the :doc:`Tutorials <Tutorials>`

* :doc:`Contributing <Contributing>`

  - Best practices and methodology for contributing to ROS 2, as well as instructions for migrating existing ROS 1 content to ROS 2

* :doc:`Contact <Contact>` 

  - Answers to your questions or a forum to start a discussion

The ROS 2 project
-----------------

If you're interested in the advancement of the ROS 2 project:

* :doc:`Releases <Releases>` 

  - Past, present and future ROS 2 distributions

* :doc:`Features Status <Features>` 

  - Features in the current release

* :doc:`Feature Ideas <Feature-Ideas>` 

>   - Ideas for nice-to-have features that are not under active development

Reviewer 2: IMO these "feature status" and "feature ideas" pages shouldn't live in the docs, but I think it's fine if we want to merge this as-is and discuss removing them + all links in a separate PR.

Reviewer: I think they do have a place here, but could be cleaned up or put in a different location.  Anyway, I think we can discuss that separately.
